bioblend/galaxy/dataset_collections/__init__.py

```diff
@@ -80,3 +94,109 @@ def __init__(self, name, id):
     "HistoryDatasetCollectionElement",
     "LibraryDatasetElement",
 )
+
+
+class DatasetCollectionClient(Client):
+
+    def __init__(self, galaxy_instance):
+        self.module = 'dataset_collections'
+        super().__init__(galaxy_instance)
+
+    def show_dataset_collection(self, dataset_collection_id: str,
+                                instance_type: str = 'history') -> dict:
+        """
+        Get details of a given dataset collection of the current user
+
+        :type dataset_collection_id: str
+        :param dataset_collection_id: dataset collection ID
+
+        :type instance_type: str
+        :param instance_type: instance type of the collection - 'history' or 'library'
+
+        :rtype: dict
+        :return: element view of the dataset collection
+        """
+        params = {
+            'id': dataset_collection_id,
+            'instance_type': instance_type,
+        }
+        url = self._make_url(module_id=dataset_collection_id)
+        return self._get(url=url, params=params)
+
+    def download_dataset_collection(self, dataset_collection_id: str, file_path: str,
+                                    history_id: Optional[str] = None):
+        """
+        Download a history dataset collection as a tgz archive.
+
+        :type dataset_collection_id: str
+        :param dataset_collection_id: Encoded dataset collection ID
+
+        :type file_path: str
+        :param file_path: The path to which the archive will be downloaded
+
+        :type history_id: str
+        :param history_id: Encoded ID of the collection's history
+
+        .. note::
+          This method is only supported by Galaxy 18.01 or later.
+        """
+        url = self._make_url(module_id=dataset_collection_id) + '/download'
+        params = {'history_id': history_id} if history_id else None
+        r = self.gi.make_get_request(url, stream=True, params=params)
+        r.raise_for_status()
+
+        with open(file_path, 'wb') as fp:
+            for chunk in r.iter_content(chunk_size=CHUNK_SIZE):
+                if chunk:
+                    fp.write(chunk)
+
+    def wait_for_dataset_collection(self, dataset_collection_id: str, maxwait: float = 12000,
+                                    interval: float = 3, proportion_complete: float = 1.0,
+                                    check: bool = True) -> dict:
+        """
+        Wait until a dataset is in a terminal state.
+
+        :type dataset_id: str
```

Comment:
dataset_collection_id

Reply:
Can you update the rest of the docstring please and add the proportion_complete argument?